autoscoper/src/ui/AutoscoperMainWindow.cpp

```diff
@@ -2438,9 +2438,53 @@ void AutoscoperMainWindow::key_k_pressed()
 {
   ui->toolButtonNextCurveSet->click();
 }
-/*void AutoscoperMainWindow::key_p_pressed(){
-   ui->toolButtonRetrack->click();
-   } */
+void AutoscoperMainWindow::key_p_pressed()
+{
+  // Get the current position of the manipulator and the volume
+  Mat4d manip = getManipulator(-1)->transform();
+  CoordFrame vol_world = (CoordFrame::from_matrix(trans(manip)) * *tracker->trial()->getVolumeMatrix(-1));
+
+  // The translation center is the relative position of the center of the volume to the corner coordinate.
+  // Since this describes the movement from the corner to the center we transform the inverse of this point
+  // by the volumes R and t to get the position of the manipular in world space.
+  Vec3f trans_center = tracker->getVolumeDescription(-1)->transCenterVec();
```

Comment:
For now, I suggest to avoid adding the new API `transCenterVec`, instead consider explicitly converting to `Vec3f` vector like is done in `key_o_pressed`